tests/bin/setup.mysql.bat

```diff
@@ -43,6 +43,7 @@ call %DIR%\base.bat :check
 if "%errorlevel%" == "1" exit /B 1
 
 "%mysql%" --host="%DB_HOSTNAME%" -u"%DB_USER%" %pw_option% -e ^"^
+SET GLOBAL sql_mode=(SELECT REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY',''));^
```

Comment:
It solves the BC-problem for tests, but not for the Propel users, right?

Reply:
Users have to do that themselves for years.
We can go away from MySQL specific Hacks in follow up PRs maybe.

Reply:
I spent some time debating this but came to this decision after looking at the matrix of options

<table><thead><tr><th>Does upgrading <br>cause a major?</th><th>Add In Tests</th><th>Add in Connection<br></th><th>Make code work without it</th></tr></thead><tbody><tr><td>Customer Has <br>ONLY_FULL_GROUP_BY<br>in project</td><td>No</td><td>No</td><td>Yes</td></tr><tr><td>Customer<br>Does not have<br>ONLY_FULL_GROUP_BY<br>in project</td><td>No</td><td>Yes</td><td>Maybe</td></tr></tbody></table>

I think this feature should get deprecated and replaced in the future, and I'm in favour of adding a new adapter to support modern MySQL (there was some discussion about it in old issue tickets a few years ago). For the time being, I think this is the best choice to stabilise the existing system and behaviours.